traits/ctraits.c

```diff
@@ -5572,5 +5573,96 @@ PyInit_ctraits(void)
     }
     Py_DECREF(trait_errors);
 
+    /* Export default-value constants, so that they can be re-used in
+       the DefaultValue enumeration. */
+    error = PyModule_AddIntConstant(
+        module,
+        "_CONSTANT_DEFAULT_VALUE",
+        CONSTANT_DEFAULT_VALUE
+    );
+    if (error < 0) {
+        return NULL;
+    }
+    error = PyModule_AddIntConstant(
+        module,
+        "_MISSING_DEFAULT_VALUE",
+        MISSING_DEFAULT_VALUE
+    );
+    if (error < 0) {
+        return NULL;
+    }
+    error = PyModule_AddIntConstant(
+        module,
+        "_OBJECT_DEFAULT_VALUE",
+        OBJECT_DEFAULT_VALUE
+    );
+    if (error < 0) {
+        return NULL;
+    }
+    error = PyModule_AddIntConstant(
+        module,
+        "_LIST_COPY_DEFAULT_VALUE",
+        LIST_COPY_DEFAULT_VALUE
+    );
+    if (error < 0) {
+        return NULL;
+    }
+    error = PyModule_AddIntConstant(
+        module,
+        "_DICT_COPY_DEFAULT_VALUE",
+        DICT_COPY_DEFAULT_VALUE
+    );
+    if (error < 0) {
+        return NULL;
+    }
+    error = PyModule_AddIntConstant(
+        module,
+        "_TRAIT_LIST_OBJECT_DEFAULT_VALUE",
+        TRAIT_LIST_OBJECT_DEFAULT_VALUE
+    );
+    if (error < 0) {
+        return NULL;
+    }
+    error = PyModule_AddIntConstant(
+        module,
+        "_TRAIT_DICT_OBJECT_DEFAULT_VALUE",
+        TRAIT_DICT_OBJECT_DEFAULT_VALUE
+    );
+    if (error < 0) {
+        return NULL;
+    }
+    error = PyModule_AddIntConstant(
+        module,
+        "_TRAIT_SET_OBJECT_DEFAULT_VALUE",
+        TRAIT_SET_OBJECT_DEFAULT_VALUE
+    );
+    if (error < 0) {
+        return NULL;
+    }
+    error = PyModule_AddIntConstant(
+        module,
+        "_CALLABLE_DEFAULT_VALUE",
+        CALLABLE_DEFAULT_VALUE
+    );
+    if (error < 0) {
+        return NULL;
+    }
+    error = PyModule_AddIntConstant(
+        module,
+        "_CALLABLE_AND_ARGS_DEFAULT_VALUE",
+        CALLABLE_AND_ARGS_DEFAULT_VALUE
+    );
+    if (error < 0) {
+        return NULL;
+    }
```

Comment:
[OCD] I noticed that the ordering of the above three constants match the ordering in `DefaultValue` but not in the ordering in `ctraits.c`. :( [/OCD]

Reply:
Yes, that was deliberate: it  seemed to make sense to keep the set/list/dict variants together.